lobsterpy/plotting/__init__.py

```diff
@@ -226,3 +236,427 @@ def _broaden(energies: np.ndarray, population: np.ndarray, sigma=None, cutoff=4.
+        fig.update_layout(legend=ld.legend_style_dict)
+
+        # Add all traces to figure
+        for _, val_trace in traces.items():
+            for trace in val_trace:
+                fig.add_trace(trace)
+
+        # Update layout with dropdown menu if label resolved plot
+        if len(traces) > 2:
+            # Update visibility of traces
+            for i, _ in enumerate(fig.data):
+                if i <= len(traces["All"]) - 1:
+                    pass
+                else:
+                    fig.data[i].visible = False
+            # Add dropdown buttons
+            fig.update_layout(
+                updatemenus=[
+                    {
+                        "buttons": [
+                            {
+                                "args": [
+                                    {
+                                        "visible": [
+                                            selected_group == group
+                                            for group, val_trace in traces.items()
+                                            for _trace in val_trace
+                                        ]
+                                    }
+                                ],
+                                "label": selected_group,
+                                "method": "update",
+                            }
+                            for selected_group in traces.keys()
+                        ],
+                        "direction": "down",
+                        "showactive": True,
+                        "active": 0,
+                        "x": 0.5,
+                        "y": 1.15,
+                        "bgcolor": "rgba(255,255,255,0.8)",
+                        "bordercolor": "rgba(0,0,0,0.2)",
+                        "xanchor": "center",
+                        "yanchor": "top",
+                        "font": {"family": "Arial", "color": "#444444", "size": 18},
+                    }
+                ]
+            )
+
+        if xlim:
+            fig.update_xaxes(range=xlim)
+        if ylim:
+            fig.update_yaxes(range=ylim)
+
+        fig.update_yaxes(automargin=True)
+
+        return fig
+
+    @staticmethod
+    def _insert_number_of_bonds_in_label(
+        label: str, character: str, number_of_bonds: int
+    ):
+        """
+        Adds number of bonds to bond label.
+        For example : for input label 'Ba1: Ba-Ti', character ':', number_of_bonds: 3,
+        Will return 'Ba1: 4 x Ba-Ti'
+
+        Args:
+            label: bond label to which number of bonds needs to be inserted
+            character: string character where number of bonds needs to be inserted
+            number_of_bonds: number of bonds corresponding to the label
+
+        Returns:
+             bond label with number of bonds inserted
+        """
+        str_number_of_bonds = str(number_of_bonds) + " x"
+        return label.replace(character, character + " " + str_number_of_bonds, 1)
```

Comment:
```python
return label.replace(character, f"{character} {number_of_bonds} x", 1)
```

seems a little easier to read.

Reply:
Thanks! I made the requested change here now. I will adapt this approach in other parts of code as well when working on documentation update 😃